### PR TITLE
Remove webpack for functions deployment

### PR DESCRIPTION
**Pull request by echaidemenos**
This PR removes the need of `webpack` for the firebase functions deployment.
Webpack 4 is not supported by node 18 and webpack 5 introduced some issues to our already "hacky" building solution.
Now the build is more straightforward using just `tsc`.

### CODE REVIEW

File: firebase.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "functions": {
+    "source": ".",
+    "codebase": "default",
+    "ignore": [
```
Comment (ericboucher) [inline firebase.json:5]: Could we use an "incldue" logic instead?

**Comment by ericboucher** [comment]
It seems odd that we actually have to move everything to the top level. Was there no way to keep this in an independent folder?

File: packages/api/.env.example [hunk 2]
```diff
@@ -5,9 +5,9 @@ POSTGRES_HOST=localhost
 POSTGRES_PORT=54321
 POSTGRES_DATABASE=postgres
 TEST_POSTGRES_DATABASE=test_ovio
-FIREBASE_PROJECT_ID=
-FIREBASE_PRIVATE_KEY=
-FIREBASE_CLIENT_EMAIL=
+FB_PROJECT_ID=
```
Comment (ericboucher) [inline packages/api/.env.example:8]: why this change?

Comment (echaidemenos) [reply packages/api/.env.example:8]: This was done due to this note: https://firebase.google.com/docs/functions/config-env?gen=2nd#reserved-names
but it looks like Firebase is not complaining, maybe because the `.env` is under `./packages/api` and not under `./` (where `firebase.json` is located).
I will be reverting the change.

Comment (ericboucher) [reply packages/api/.env.example]: firebase is not using the .env, we are setting the environment variables through `config:cloud-functions`

File: packages/api/cloud-functions/index.ts [hunk 3]
```diff
@@ -14,10 +14,47 @@ import { sendSlackMessage } from '../src/utils/slack.utils';
 import { checkBuoysStatus } from '../src/workers/check-buoys-status';
 import { dataSourceOptions } from '../ormconfig';
 
-// We have to manually import all required entities here, unfortunately - the globbing that is used in ormconfig.ts
-// doesn't work with Webpack. This declaration gets processed by a custom loader (`add-entities.js`) to add import
-// statements for all the entity classes.
-// import-all-entities
+import * as SiteAuditEntity from '../src/audit/site-audit.entity';
```
Comment (ericboucher) [inline packages/api/cloud-functions/index.ts:17]: Can't we keep using a formulae like we had in `[add-entities.js](https://github.com/aqualinkorg/aqualink-app/pull/866/files#diff-7f237b5a6c3cf6ad00f2b425ac060371215e0c3f6f6ec9e59395ada53b867d37)`?

File: packages/website/package.json [hunk 4]
```diff
@@ -91,7 +91,7 @@
     "@types/immutable": "^3.8.7",
     "@types/jest": "^24.0.0",
     "@types/lodash": "^4.14.149",
-    "@types/node": "^20.2.3",
+    "@types/node": "^18.2.3",
```
Comment (ericboucher) [inline packages/website/package.json:94]: I think [18.16.16](https://www.npmjs.com/package/@types/node/v/18.16.16) is the latest version

File: packages/api/cloud-functions/index.ts (hunk 3 above)
Comment (ericboucher) [reply packages/api/cloud-functions/index.ts:17]: I'd much rather have it automated and not have to think about exporting entities here whenever we add a new one.

File: firebase.json (hunk 1 above)
Comment (echaidemenos) [reply firebase.json:5]: unfortunately there isn't

File: packages/api/cloud-functions/index.ts (hunk 3 above)
Comment (echaidemenos) [reply packages/api/cloud-functions/index.ts:17]: I have added some per-prossesing, almost identical to what we did before, let me know what you think

File: package.json [hunk 5]
```diff
@@ -46,7 +47,7 @@
     "typescript": "^4.5.5"
   },
   "engines": {
-    "node": "18.x",
+    "node": "18",
```
[the remark this answers is not in the document]
Comment (echaidemenos) [reply package.json:50]: Hello! That is not true as far as i know and tested. `18` will match any node 18 version, just as `18.x` would do. The change here is because firebase requires the simple `18` format: https://firebase.google.com/docs/functions/manage-functions?gen=2nd#set_nodejs_version
